include/libobmcsession/manager.hpp

```diff
@@ -0,0 +1,192 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (C) 2021 YADRO.
+ */
+
+#pragma once
+
+#include <sdbusplus/bus.hpp>
+#include <xyz/openbmc_project/Session/Item/server.hpp>
+
+namespace obmc
+{
+namespace session
+{
+
+using SessionItemServer =
+    sdbusplus::xyz::openbmc_project::Session::server::Item;
+
+class SessionManager;
+class SessionItem;
+
+using SessionItemUni = std::unique_ptr<SessionItem>;
+using SessionManagerPtr = std::shared_ptr<SessionManager>;
+using SessionManagerWeakPtr = std::weak_ptr<SessionManager>;
+
+class SessionManager final : public std::enable_shared_from_this<SessionManager>
+{
+    static constexpr const char* serviceNameStartSegment =
+        "xyz.openbmc_project.Session.";
+
+  public:
+    using SessionIdentifier = std::uint64_t;
+    using SessionType = SessionItemServer::Type;
+    using SessionCleanupFn = std::function<bool(SessionIdentifier)>;
+
+    SessionManager() = delete;
+    ~SessionManager() = default;
+    SessionManager(const SessionManager&) = delete;
+    SessionManager& operator=(const SessionManager&) = delete;
+    SessionManager(SessionManager&&) = delete;
+    SessionManager& operator=(SessionManager&&) = delete;
+
+    /** @brief Constructs session manager
+     *
+     * @param[in] bus     - Handle to system dbus
+     * @param[in] objPath - The Dbus service slug uniquely identifies the source
+     *                      of a session to created appropriate items from.
+     *                      Service name template:
+     *                      'xyz.openbmc_project.Session.${slug}'.
+     * @param[in] type    - Type of all session items which will be created by
+     *                      the current instance.
+     */
+    SessionManager(sdbusplus::bus::bus& bus, const std::string& slug,
+                   const SessionType type) :
+        bus(bus),
+        slug(slug), serviceName(serviceNameStartSegment + slug), type(type)
+    {}
+
+    /**
+     * @brief Create a session and publish into the dbus.
+     *
+     * @param userName - the owner user name
+     * @param remoteAddress - the remote ip-address of session creating
+     * initiators'.
+     * @return SessionIdentifier - uniquie session ID
+     */
+    SessionIdentifier create(const std::string& userName,
+                             const uint32_t remoteAddress);
+
+    /**
+     * @brief Create a session and publish into the dbus with cleanup callback
+     * on the session destroy.
```

Comment:
Please align the second line for better readability:

```
@brief Create ...
       on the ...
```

Reply:
Fixed